File: Front_End/src/app/mg/modules/vehicule.ts
// MS Vehicule

import { TypeVehicule } from "./typeVehicule";

export class Vehicule {

  id: number;
  matricule: string;

  datePremierCirculation: Date;
  typeVehicule: TypeVehicule;

  carburant: string;
  poidsVide: number;

}

File: Back_End/sic.mg.vehicle/src/main/java/ma/sic/mg/vehicle/entities/Vehicule.java
package ma.sic.mg.vehicle.entities;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString

public class Vehicule implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String matricule;
    private double poidsVide;
    private Carburant carburant;
    private Date datePremierCirculation;

    @ManyToOne(fetch = FetchType.LAZY)
    private TypeVehicule typeVehicule;
}
